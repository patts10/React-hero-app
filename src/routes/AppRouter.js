
import { Routes, Route, BrowserRouter } from "react-router-dom";
import { DScreen } from "../components/dc/DScreen";
import { LoginScreen } from "../components/login/LoginScreen";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { SearchScreen } from "../components/search/Searchcreen";
import { Navbar } from "../components/ui/NavBar";

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Navbar />
      <Routes>
        <Route path="/" element={<MarvelScreen />} />
        <Route path="/Marvel" element={<MarvelScreen />} />
        <Route path="/dc" element={<DScreen />} />
        <Route path="/Search" element={<SearchScreen />} />
        <Route path="login" element={<LoginScreen />} />
      </Routes>
    </BrowserRouter>
  )
}
