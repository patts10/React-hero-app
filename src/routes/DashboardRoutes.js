import { Routes, Route } from "react-router-dom";

import { Navbar } from "../components/ui/NavBar"

import { DScreen } from "../components/dc/DScreen";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { SearchScreen } from "../components/search/SearchScreen";
import { HeroScreen } from "../components/hero/HeroScreen";

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />

      <div className="mx-20">
      <Routes>
        <Route path="Marvel" element={ <MarvelScreen /> } />
        <Route path="dc" element={ <DScreen /> } />

        <Route path="Search" element={ <SearchScreen /> } />
        <Route path="hero/:heroId" element={ <HeroScreen /> } />

        <Route path="/" element={ <MarvelScreen /> } />
      </Routes>
      </div>
    </>
  )
}
