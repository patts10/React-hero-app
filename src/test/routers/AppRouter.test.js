import { AppRouter } from "../../routes/AppRouter";
import { mount } from 'enzyme'
import { AuthContext } from "../../auth/authContext";



describe('Pruebas en <AppRouter />', () => { 
  
  test('should show login view if user is not authenticated', () => { 
    
    const contextValue = {
      user: {
        logged: false
      }
    };
    const wrapper = mount(
      <AuthContext.Provider value={ contextValue } >
        <AppRouter />
      </AuthContext.Provider>
      );

      expect( wrapper ).toMatchSnapshot();
      expect( wrapper.find('.login').text().trim() ).toBe( 'Login' );

   });
  
  test('should show Marvel component if user is authenticated', () => { 
    
    const contextValue = {
      user: {
        logged: true,
        name: 'Patts'
      }
    };
    const wrapper = mount(
      <AuthContext.Provider value={ contextValue } >
        <AppRouter />
      </AuthContext.Provider>
      );

      expect( wrapper ).toMatchSnapshot();
      expect( wrapper.find('.navbar').exists() ).toBe( true );

   });

 });