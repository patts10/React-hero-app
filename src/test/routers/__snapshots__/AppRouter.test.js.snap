// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> should show Marvel component if user is authenticated 1`] = `
Array [
  <nav
    className="relative w-full flex flex-wrap items-center justify-between py-3 bg-gray-900 text-gray-200 shadow-lg navbar navbar-expand-lg navbar-light"
  >
    <div
      className="container-fluid w-full flex flex-wrap items-center justify-between px-6"
    >
      <div
        className="collapse navbar-collapse flex flex-row items-center w-full"
      >
        <a
          className="text-xl text-white pr-2 font-semibold"
          href="/"
          onClick={[Function]}
        >
          Asociaciones
        </a>
        <ul
          className="navbar-nav flex flex-row pl-0 list-style-none mr-auto w-full"
        >
          <li
            className="nav-item p-2"
          >
            <a
              aria-current="page"
              className="nav-item active"
              href="/marvel"
              onClick={[Function]}
            >
              Marvel
            </a>
          </li>
          <li
            className="nav-item p-2"
          >
            <a
              className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
              href="/dc"
              onClick={[Function]}
            >
              Dc
            </a>
          </li>
          <li
            className="nav-item p-2"
          >
            <a
              className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
          <li
            className="nav-item p-2 absolute right-0 mx-5"
          >
            <span
              className="text-sky-600 mx-3 text-info"
            >
              Patts
            </span>
            <button
              className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
              onClick={[Function]}
            >
              Logout
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="mx-20"
  >
    <div
      className="my-5"
    >
      <h1
        className="text-3xl"
      >
        MarvelScreen
      </h1>
      <hr />
      <div
        className="flex justify-around flex-wrap my-3 animate__animated animate__fadeIn"
      >
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Spider Man"
                src="assets/marvel-spider.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Spider Man
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Peter Parker
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Captain America"
                src="assets/marvel-captain.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Captain America
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Steve Rogers
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Iron Man"
                src="assets/marvel-iron.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Iron Man
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Tony Stark
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Thor"
                src="assets/marvel-thor.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Thor
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Thor Odinson
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Hulk"
                src="assets/marvel-hulk.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Hulk
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Bruce Banner
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Wolverine"
                src="assets/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Wolverine
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  James Howlett
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Daredevil"
                src="assets/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Daredevil
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Matthew Michael Murdock
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    Daredevil #1
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Hawkeye"
                src="assets/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Hawkeye
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Clinton Francis Barton
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Cyclops"
                src="assets/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Cyclops
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Scott Summers
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    X-Men #1
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rounded-lg border border-gray-300 mb-4 text-center m-3 lg:w-3/12 md:w-2/5 animate__animated animate__fadeIn"
        >
          <div
            className="flex flex-row"
          >
            <div
              className="w-4/12"
            >
              <img
                alt="Silver Surfer"
                src="assets/marvel-silver.jpg"
              />
            </div>
            <div
              className="flex items-center w-8/12"
            >
              <div
                className="py-4 w-full text-left px-3"
              >
                <h5
                  className="md:text-2xl sm:text-2xl"
                >
                  Silver Surfer
                </h5>
                <p
                  className="md:text-base sm:text-sm mt-3"
                >
                  Norrin Radd
                </p>
                <p>
                  <small
                    className="text-gray-600"
                  >
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  className="hover:underline hover:text-blue-900"
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  more...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <AppRouter /> should show login view if user is not authenticated 1`] = `
<section
  className="h-screen"
>
  <div
    className="px-6 h-full text-gray-800"
  >
    <div
      className="flex xl:justify-center lg:justify-between justify-center items-center flex-wrap h-full g-6"
    >
      <div
        className="grow-0 shrink-1 md:shrink-0 basis-auto xl:w-6/12 lg:w-6/12 md:w-9/12 mb-12 md:mb-0"
      >
        <img
          alt="Sample image"
          className="w-full"
          src="https://mdbcdn.b-cdn.net/img/Photos/new-templates/bootstrap-login-form/draw2.webp"
        />
      </div>
      <div
        className="xl:ml-20 xl:w-5/12 lg:w-5/12 md:w-8/12 mb-12 md:mb-0"
      >
        <form>
          <div
            className="flex flex-row items-center justify-center lg:justify-start"
          >
            <p
              className="text-lg mb-0 mr-4"
            >
              Sign in with
            </p>
            <button
              className="inline-block p-3 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded-full shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out mx-1"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              <svg
                className="w-4 h-4"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              className="inline-block p-3 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded-full shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out mx-1"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              <svg
                className="w-4 h-4"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              className="inline-block p-3 bg-blue-600 text-white font-medium text-xs leading-tight uppercase rounded-full shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out mx-1"
              data-mdb-ripple="true"
              data-mdb-ripple-color="light"
              type="button"
            >
              <svg
                className="w-4 h-4"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
          <div
            className="flex items-center my-4 before:flex-1 before:border-t before:border-gray-300 before:mt-0.5 after:flex-1 after:border-t after:border-gray-300 after:mt-0.5"
          >
            <p
              className="text-center font-semibold mx-4 mb-0"
            >
              Or
            </p>
          </div>
          <div
            className="mb-6"
          >
            <input
              className="form-control block w-full px-4 py-2 text-xl font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
              id="emailInput"
              placeholder="Email address"
              type="text"
            />
          </div>
          <div
            className="mb-6"
          >
            <input
              className="form-control block w-full px-4 py-2 text-xl font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
              id="passwordInput"
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            className="flex justify-between items-center mb-6"
          >
            <div
              className="form-group form-check"
            >
              <input
                className="form-check-input appearance-none h-4 w-4 border border-gray-300 rounded-sm bg-white checked:bg-blue-600 checked:border-blue-600 focus:outline-none transition duration-200 mt-1 align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer"
                id="exampleCheck2"
                type="checkbox"
              />
              <label
                className="form-check-label inline-block text-gray-800"
                htmlFor="exampleCheck2"
              >
                Remember me
              </label>
            </div>
            <a
              className="text-gray-800"
              href="#!"
            >
              Forgot password?
            </a>
          </div>
          <div
            className="text-center lg:text-left"
          >
            <button
              className="login inline-block px-7 py-3 bg-blue-600 text-white font-medium text-sm leading-snug uppercase rounded shadow-md hover:bg-blue-700 hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out"
              onClick={[Function]}
              type="button"
            >
              Login
            </button>
            <p
              className="text-sm font-semibold mt-2 pt-1 mb-0"
            >
              Don't have an account?
              <a
                className="text-red-600 hover:text-red-700 focus:text-red-700 transition duration-200 ease-in-out"
                href=""
              >
                Register
              </a>
            </p>
          </div>
        </form>
      </div>
    </div>
  </div>
</section>
`;
